### PR TITLE
Np 1347/reverse google sheets integration column header to prop mapper

### PR DESCRIPTION
**Pull request by olhabn**
ticket: https://nanopay.atlassian.net/browse/NP-1347
nanopay pr: https://github.com/nanoPayinc/NANOPAY/pull/10105

@kgr @jlhughes please review

### CODE REVIEW

File: src/foam/u2/detail/SectionedDetailPropertyView.js [hunk 1]
```diff
@@ -279,8 +279,8 @@ foam.CLASS({
                     .end()
                   .end();
                 })
-              .end()
-              .callIf(prop.validationTextVisible && mode === self.DisplayMode.RW, function() {
```
Comment (olhabn) [inline src/foam/u2/detail/SectionedDetailPropertyView.js:274]: is there any chance we could do such a change

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      hidden: true,
+      expression: function(googleSheetLink) {
+        var findMatch = googleSheetLink.match(this.SPREADSHEET_ID_REGEX);
+        return  findMatch ? findMatch[1] : findMatch;
+      }
+    },
+    {
+      name: 'googleSheetId',
+      class: 'String',
+      view: {
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: You could just do: view: 'foam.u2.TextField'.

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        class: 'foam.u2.TextField',
+      },
+      label: 'Google Sheets Id',
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: Won't this be the default label anyway?

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      class: 'String',
+      hidden: true
+    },
+    {
+      name: 'googleSpreadsheetId',
+      class: 'String',
+      view: {
+        class: 'foam.u2.TextField',
+      },
+      label: 'Google Spreadsheet Id',
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: Isn't this already the default?

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    {
+      name: 'importClassInfo',
+      class: 'Class',
+      javaType: 'foam.core.ClassInfo',
+      hidden: true,
+    },
+    {
+      name: 'googleSheetLink',
+      class: 'String',
+      view: {
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: Here and elsewhere, you can just do: view: 'foam.u2.TextField'.

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsTransactionsDataImportServiceImpl.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/GoogleSheetsTransactionsDataImportServiceImpl.js:7]: Add copyright header.

File: src/foam/nanos/google/api/sheets/import/ImportFromGoogleSheetsModal.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/ImportFromGoogleSheetsModal.js]: copyright

File: src/foam/nanos/google/api/sheets/import/views/modal/ColumnsToPropertiesMappingModal.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/views/modal/ColumnsToPropertiesMappingModal.js]: copyright

File: src/foam/nanos/google/api/sheets/import/views/modal/GoogleSheetImportModal.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
+foam.CLASS({
+  package: 'foam.nanos.google.api.sheets',
+  name: 'GoogleSheetImportModal',
+  extends: 'net.nanopay.ui.wizardModal.WizardModalSubView',
+  imports: [
+    'importConfig',
+    'importServiceName',
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/views/modal/GoogleSheetImportModal.js:17]: no trailing commas

File: src/foam/nanos/google/api/sheets/import/views/modal/ImportFromGoogleSheetsForm.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    },
+    'of',
+    'dao'
+  ],
+  methods: [
+    function init() {
+      this.SUPER();
+      // var self = this;
+      this.views = {
+        'googleSheetLink'        : { view: { class: 'foam.nanos.google.api.sheets.GoogleSheetImportModal' }, startPoint: true },
```
Comment (kgrgreer) [inline src/foam/nanos/google/api/sheets/import/views/modal/ImportFromGoogleSheetsForm.js:43]: simplify view:, next line also

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js (hunk 4 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: fixed

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsTransactionsDataImportServiceImpl.js (hunk 6 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/GoogleSheetsTransactionsDataImportServiceImpl.js:7]: fixed

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js (hunk 2 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: fixed

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js (hunk 3 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: fixed

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js (hunk 5 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/GoogleSheetsImportConfig.js]: fixed

File: src/foam/nanos/google/api/sheets/import/ImportFromGoogleSheetsModal.js (hunk 7 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/ImportFromGoogleSheetsModal.js]: fixed

File: src/foam/nanos/google/api/sheets/import/views/modal/ColumnsToPropertiesMappingModal.js (hunk 8 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/views/modal/ColumnsToPropertiesMappingModal.js]: fixed

File: src/foam/nanos/google/api/sheets/import/views/modal/GoogleSheetImportModal.js (hunk 9 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/views/modal/GoogleSheetImportModal.js:17]: fixed

File: src/foam/nanos/google/api/sheets/import/views/modal/ImportFromGoogleSheetsForm.js (hunk 10 above)
Comment (olhabn) [reply src/foam/nanos/google/api/sheets/import/views/modal/ImportFromGoogleSheetsForm.js:43]: for some reason it doesn't work for me

File: src/foam/nanos/google/api/sheets/import/GoogleSheetsDataImportServiceImpl.js [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+      javaCode: `
+        GoogleSheetsApiService googleSheetsAPIEnabler = (GoogleSheetsApiService)x.get("googleSheetsDataExport");
+        ValueRange values;
+        ImportDataMessage result = new ImportDataMessage();
+        try {
+          values = googleSheetsAPIEnabler.getFormatedValues(x, importConfig.getGoogleSpreadsheetId(), importConfig.getGoogleSheetId());
+          
+          importConfig.setCellsRange(values.getRange().split("!")[1]);
+
+          List<List<Object>> data = values.getValues();
```
Comment (olhabn) [inline src/foam/nanos/google/api/sheets/import/GoogleSheetsDataImportServiceImpl.js:113]: suggestion: 
if there was a problem with adding records we still might want to update user's google sheet with ids or status
so we might need extra logic for adding status that might be too much thought

File: src/foam/core/Property.js [hunk 12]
```diff
@@ -293,6 +293,11 @@ foam.CLASS({
       name: 'sortable',
       value: true
     },
+    {
+      class: 'Boolean',
+      name: 'sheetsOutput',
+      value: false
```
Comment (kgrgreer) [inline src/foam/core/Property.js]: false is already the default default, so this line can be removed

Comment (olhabn) [reply src/foam/core/Property.js]: fixed

**Pull request merged**